source/renderer/app/config/epochsConfig.js

```diff
@@ -8,29 +8,27 @@ export const START_TIME_STAGING = 1506450213;
 export const START_TIME_TESTNET = 1537941600;
 export const START_TIME_DEVELOPMENT = 1541808003;
 
-export const SLOT_DURATION = 1; // unit: seconds
-export const SLOT_DURATION_STN = 1; // unit: seconds
-export const SLOT_DURATION_ITN = 2; // unit: seconds
+export const DEFAULT_SLOT_LENGTH = { quantity: 1, unit: 'second' };
+export const DEFAULT_SLOT_LENGTH_STN = { quantity: 1, unit: 'second' };
+export const DEFAULT_SLOT_LENGTH_ITN = { quantity: 2, unit: 'second' };
 
-export const TOTAL_SLOTS = 432000;
-export const TOTAL_SLOTS_STN = 21600;
-export const TOTAL_SLOTS_ITN =
-  isDev || isTest || isIncentivizedTestnetSelfnode ? 150 : 43200;
+export const DEFAULT_EPOCH_LENGTH = { quantity: 432000, unit: 'slot' };
+export const DEFAULT_EPOCH_LENGTH_STN = { quantity: 21600, unit: 'slot' };
+export const DEFAULT_EPOCH_LENGTH_ITN =
+  isDev || isTest || isIncentivizedTestnetSelfnode
+    ? { quantity: 150, unit: 'slot' }
+    : { quantity: 43200, unit: 'slot' };
 
-export const EPOCH_LENGTH = SLOT_DURATION * TOTAL_SLOTS; // 5 days | unit: seconds
-export const EPOCH_LENGTH_STN = SLOT_DURATION_STN * TOTAL_SLOTS_STN; // 6 hours | unit: seconds
-export const EPOCH_LENGTH_ITN = SLOT_DURATION_ITN * TOTAL_SLOTS_ITN; // 1 day / 5 minutes (isDev || isTest || isIncentivizedTestnetSelfnode = true) | unit: seconds
+export const DEFAULT_EPOCH_COUNTDOWN_INTERVAL = 1 * 1000; // 1 second | unit: milliseconds;
 
-export const EPOCH_COUNTDOWN_INTERVAL = 1 * 1000; // 1 second | unit: milliseconds;
-
-export const getTotalSlots = () => {
-  if (isIncentivizedTestnet) return TOTAL_SLOTS_ITN;
-  if (isShelleyTestnet) return TOTAL_SLOTS_STN;
-  return TOTAL_SLOTS;
+export const getDefaultSlotLength = () => {
+  if (isIncentivizedTestnet) return DEFAULT_SLOT_LENGTH_ITN;
+  if (isShelleyTestnet) return DEFAULT_SLOT_LENGTH_STN;
+  return DEFAULT_SLOT_LENGTH;
 };
 
-export const getEpochLength = () => {
-  if (isIncentivizedTestnet) return EPOCH_LENGTH_ITN;
-  if (isShelleyTestnet) return EPOCH_LENGTH_STN;
-  return EPOCH_LENGTH;
+export const getDefaultEpochLength = () => {
```

Comment:
@yakovkaravelov can't we eliminate the need for these hardcoded values? This is not safe... We should only use the ones we get from the Api.

Reply:
@nikolaglumac This is just initialization part before api returns response.
As soon as API returns values we use them instead of these values.

Reply:
Yes but this could result in false data - e.g. if epoch length changes - it can change dynamically.
I would prefer to remove any default data and have Daedalus not show parts of the screen where this data is used until it is loaded from the Api!

Reply:
@nikolaglumac Fixed this. Thanks.